File: SourceCode/WebSite/API Services/BusinessObjects/ReactionData.vb
#Region " Imports "

Imports Newtonsoft.Json.Linq ' OpenSource

#End Region

''' <summary>
''' Reaction Data
''' </summary>
''' <remarks></remarks>
Public Class ReactionData

#Region " Public Properties "

    Public Property ReactionMedDrapt As String
    Public Property ReactionMeddraversionPt As String
    Public Property ReactionOutcome As EnumReactionOutcome

#End Region

#Region " Public Methods "

    ''' <summary>
    ''' Convert JSON Data
    ''' </summary>
    ''' <param name="jToken">Token</param>
    ''' <returns>List of Reaction Data</returns>
    ''' <remarks></remarks>
    Public Shared Function ConvertJsonData(jToken As JToken) As List(Of ReactionData)

        Dim data As New List(Of ReactionData)

        If IsJTokenValid(jToken) Then

            For Each reaction In jToken

                Dim obj As New ReactionData

                obj.ReactionMedDrapt = reaction("reactionmeddrapt")
                obj.ReactionMeddraversionPt = reaction("reactionmeddraversionpt")
                Integer.TryParse(reaction("reactionoutcome"), obj.ReactionOutcome)

                data.Add(obj)

            Next

        End If

        Return data

    End Function

#End Region

End Class

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/ReactionData.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;
using ShopAware.Core.Enumerations;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        /// <summary>
        ///     Reaction Data
        /// </summary>
        /// <remarks></remarks>
        public class ReactionData
        {
            #region Properties

            public string ReactionMedDrapt { get; set; }

            public string ReactionMeddraversionPt { get; set; }

            public ReactionOutcome ReactionOutcome { get; set; }

            #endregion

            #region Public Methods

            /// <summary>
            ///     Convert JSON Data
            /// </summary>
            /// <param name="jToken">Token</param>
            /// <returns>List of Reaction Data</returns>
            /// <remarks></remarks>
            public static List<ReactionData> ConvertJsonData(JToken jToken)
            {
                var data = new List<ReactionData>();

                if (Utilities.IsJTokenValid(jToken))
                {
                    foreach (var reaction in jToken)
                    {
                        var obj = new ReactionData();

                        obj.ReactionMedDrapt = Utilities.GetJTokenString(reaction, "reactionmeddrapt");
                        obj.ReactionMeddraversionPt = Utilities.GetJTokenString(reaction, "reactionmeddraversionpt");

                        var reactionOutcome = (int) obj.ReactionOutcome;

                        int.TryParse((Utilities.GetJTokenString(reaction, "reactionoutcome")), out reactionOutcome);
                        obj.ReactionOutcome = (ReactionOutcome) reactionOutcome;

                        data.Add(obj);
                    }
                }

                return data;
            }

            #endregion
        }
    }
}